src/components/02_atoms/MultipleFields/MultipleFields.js

```diff
@@ -0,0 +1,206 @@
+import ListItemText from '@material-ui/core/ListItemText';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
+
+const Add = styled('div')`
+  .icon {
+    margin-left: 10px;
+  }
+`;
+
+const style = {
+  listItemStyles: {
+    paddingLeft: 0,
+  },
+  ListItemIconStyles: {
+    cursor: 'move',
+  },
+};
+
+class MultipleFields extends Component {
+  static propTypes = {
+    label: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired,
+    component: PropTypes.func.isRequired,
+    value: PropTypes.arrayOf(PropTypes.string).isRequired,
+  };
+
+  /**
+   * Initial state
+   */
+  state = {
+    currentIndex: -1,
+    selectedValue: null,
+  };
+
+  componentDidMount() {
+    const { value, onChange } = this.props;
+    // Set the initial value to an empty string if no values in array.
+    if (!value.length) {
+      onChange(['']);
+    }
+  }
+
+  /**
+   * Sets the state value with the selected element.
+   * @param {Event} event
+   */
+  onDragStart = event => {
+    // setData needed for FireFox, needs to setData to work
+    event.dataTransfer.setData('text', '');
+    event.dataTransfer.effectAllowed = 'move';
+    const currentIndex = parseInt(event.currentTarget.dataset.key, 10);
+
+    this.setState({
+      currentIndex,
+      selectedValue: this.props.value[currentIndex],
+    });
+  };
+
+  /**
+   * Changes the current value with the value that is under the
+   * current value, and replace the over value with the current
+   * value.
+   * @param {Event} event
+   */
+  onDragOver = event => {
+    event.preventDefault();
+    const {
+      props: { value },
+      state: { currentIndex, selectedValue },
+    } = this;
+
+    const overIndex = parseInt(event.currentTarget.dataset.key, 10);
+    if (currentIndex !== overIndex) {
+      // Changes the two values with one another
+      value[currentIndex] = value[overIndex];
+      value[overIndex] = selectedValue;
+
+      this.setState({
+        currentIndex: overIndex,
+      });
+    }
+  };
+
+  /**
+   * Will update the state and call the onChange method
+   * once the element has been reordered.
+   */
+  onDragEnd = () => {
+    const { value, onChange } = this.props;
+    this.setState(
+      {
+        currentIndex: -1,
+        selectedValue: null,
+      },
+      () => {
+        onChange(value);
+      },
+    );
+  };
+
+  /**
+   * Updated the current value of the input.
+   * @param {Event} event
+   * @param {String} value
+   */
+  changeItem = index => value => {
+    const { value: propsValue, onChange } = this.props;
+    const newValue = [...propsValue];
+    newValue[index] = value;
+    onChange(newValue);
+  };
+
+  /**
+   * Adds another empty string to the current set of values.
+   */
+  addAnotherItem = () => {
+    const { value, onChange } = this.props;
+    const newValue = [...value, ''];
```

Comment:
I'm curious, what are you thoughts of supporting other kind of widgets, like for example numbers, selected fields etc.?

Reply:
From what I can see, if it is a input type, it will be an array of string. Maybe we should do the same check as we do when creating the value for single and multiple fields: https://github.com/jsdrupal/drupal-admin-ui/blob/master/src/utils/api/schema.js#L1.